product/views/purchase_request_view.py

```
@@ -0,0 +1,101 @@
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request, item_id):
+        """
+        Handles GET requests to retrieve all purchase requests for a specific item.
+        """
+        user = request.user
+
+        try:
+            item = Item.objects.get(id=item_id, seller_user=user)
+        except Item.DoesNotExist:
+            raise ItemNotFoundException()
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)